src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -213,19 +213,20 @@ public static unsafe bool Ipv6StringToAddress(ReadOnlySpan<char> ipSpan, ushort*
                 long result = 0;
                 if (!string.IsNullOrEmpty(scopeId))
                 {
-                    if (scopeId.Length < 2)
+                    uint interfaceIndex = Interop.Sys.GetNativeIPInterfaceIndex(scopeId);
+                    if (interfaceIndex >= 0)
```

Comment:
What happens if scopeId is just a number?  Does if_nametoindex do the "right thing"?  What kind of overhead does this add?

Reply:
Changed to "call pinvoke if scopeId is not a number (and isn't empty) so it should not add any overhead.